amethyst_animation/src/resources.rs

```diff
@@ -111,6 +135,22 @@ where
             m: marker::PhantomData,
         }
     }
+
+    /// Create rest state for the hierarchy. Will copy the values from the base components for each
+    /// entity in the hierarchy.
+    pub fn rest_state<F>(&self, get_component: F, states: &mut WriteStorage<RestState<T>>)
+    where
+        T: AnimationSampling,
+        F: Fn(Entity) -> Option<T>,
+    {
+        for entity in self.nodes.values() {
+            if let None = states.get(*entity) {
```

Comment:
just a preference, but when checking for none I prefer using
if states.get(*entity).is_none()
as this shows a little bit better the intent. Just my opinion, feel free to ignore.

Reply:
I'll change it. For some reason I instinctively go to matching

Reply:
Fixed